src/test/java/teammates/test/cases/ui/browsertests/InstructorCoursesPageUiTest.java

```diff
@@ -65,6 +67,13 @@ public static void classSetup() throws Exception {
         
         /* Explanation: Gets a browser instance to be used for this class. */
         browser = BrowserPool.getBrowser();
+        
+        /* 
+         * Any entity that is created in previous test run must be deleted.
+         * If that previous test run fails, the entity persists and that will
+         * break tests.
+         */
+        BackDoor.deleteCourse(validCourse.id); // delete if it exists
```

Comment:
Consider the condition that will fail this test, as I mentioned up there. In other words, the guilty course CCAddUiTest.course1 persists after the test.
1. Under the old code
The code straightaway goes to `testContent()`. No problem in the first comparison, `coursesPage.verifyHtml("/instructorCoursesNoCourse.html")`. The second comparison, `coursesPage.verifyHtmlMainContent("/instructorCoursesMultipleCourses.html")`, because of the presence of that guilty course, will fail. And once happened, this will be _permanent_ because no part of the code up to there deletes the course.
2. Under the re-written code
Before going to test content, the guilty course (if present) will be deleted through the BackDoor operation. Thus it's the same as starting the test freshly.
The presence of that `BackDoor.deleteCourse` tells that the whoever wrote this test file before already anticipated that failure, which is good. Only that it's not placed correctly.